src/Resolver/Conditional.php

```diff
@@ -64,9 +64,14 @@ public function resolve($definition)
         }
 
         foreach ($definition->get($this->getIndicator()) as $matcher) {
-            $rawValue = $this->getIterator()->get($matcher->get('matches'));
-            $value    = is_scalar($rawValue) ? (string) $rawValue : json_encode($rawValue);
-            $pattern  = '/' . addcslashes($matcher->get('pattern'), '/') . '/';
+            try {
+                $rawValue = $this->getIterator()->get($matcher->get('matches'));
+            } catch (\Exception $exception) {
+                // exceptions thrown for match value lookups should silently fail and fall through
+                continue;
+            }
+            $value   = is_scalar($rawValue) ? (string) $rawValue : json_encode($rawValue);
```

Comment:
Should make sure to re-initialize `$rawValue` before `try`, otherwise you risk either having an undefined value here, or re-using the value from the previous loop iteration